cmd/quadlet/main.go

```diff
@@ -210,6 +211,66 @@ func enableServiceFile(outputPath string, service *parser.UnitFile) {
 	}
 }
 
+func isSha256Digest(imageName string) bool {
+	if strings.HasPrefix(imageName, "sha256:") {
+		return true
+	}
+
+	// only fully specified digests are matched
+	if len(imageName) != 64 {
+		return false
+	}
+
+	for _, c := range imageName {
+		if !unicode.Is(unicode.Hex_Digit, c) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func isFullyQualifiedName(imageName string) bool {
+	if isSha256Digest(imageName) {
```

Comment:
This is detecting strings where the _whole name_ is `sha256:…`. Not digested references like `busybox@sha256:…`. Is that intentional?

It is contrary to the documentation below.

Reply:
It is intentional, because "podman run" accepts those without doing a shortname resolve, and they seem to potentially be useful to run a fixed version. I could add it to the examples comment though.

Reply:
I think tightening the terminology a bit would be useful:
- These digest values without repo names are (when used as inputs to `podman run`) “image IDs”.
- Only names that contain repo names (`reference.Named`) could possibly be “fully qualified” or “short”